sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionContext.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Contains information about a partition that an <see cref="IPartitionProcessor" /> will
+    ///   be processing events from.
+    /// </summary>
+    ///
+    public class PartitionContext
+    {
+        /// <summary>
+        ///   The path of the specific Event Hub that the context is associated with, relative
+        ///   to the Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public string EventHubName { get; }
+
+        /// <summary>
+        ///   The name of the consumer group this context is associated with.
+        /// </summary>
+        ///
+        public string ConsumerGroup { get; }
+
+        /// <summary>
+        ///   The identifier of the associated <see cref="EventProcessor" /> instance.
+        /// </summary>
+        ///
+        public string InstanceId { get; }
```

Comment:
I'd recommend `Identifier` here for the reasons discussed in the `CheckpointManager` class.